File: C/2042-check-if-numbers-are-ascending-in-a-sentence/main.go
package main

import "fmt"

func areNumbersAscending(s string) bool {
	prev, curr := 0, 0
	for _, b := range s {
		if byte(b) >= byte('0') && byte(b) <= byte('9') {
			curr = curr*10 + int(byte(b)-byte('0'))
		} else {
			if prev != 0 && curr != 0 {
				if prev >= curr {
					return false
				}
			}
			if curr > 0 {
				prev = curr
			}
			curr = 0
		}
	}

	if prev != 0 && curr != 0 {
		if prev >= curr {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println()
}

File: C/2042-check-if-numbers-are-ascending-in-a-sentence/main.rs
struct Solution {}

impl Solution {
  pub fn are_numbers_ascending(s: String) -> bool {
    let mut prev: i32 = 0;
    let mut sum: i32 = 0;
    for b in s.as_bytes() {
      if *b < '0' as u8 || *b > '9' as u8 {
        if sum > 0 {
          if prev > 0 && sum <= prev {
            return false;
          }
          prev = sum;
          sum = 0;
        }
        continue;
      }
      sum = sum * 10 + (*b - '0' as u8) as i32;
    }
    sum == 0 || sum > 0 && sum > prev
  }
}
